torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,86 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet.ml (https://www.comet.ml/).
+
+    Args:
+        project_name (Optional[str]): Comet.ml project name. Defaults to Uncategorized.
+        workspace (Optional[str]): Comet.ml workspace name. If not provided, uses the default workspace.
+        experiment_name (Optional[str]): Name of the experiment. If not provided, Comet will auto-generate a name.
+        tags (Optional[List[str]]): Tags to associate with the experiment.
+        log_code (bool): Whether to log the source code. Defaults to True.
+        **kwargs: Additional arguments to pass to comet_ml.Experiment.
+
+    Example:
+        >>> from torchtune.utils.metric_logging import CometLogger
+        >>> logger = CometLogger(project_name="my_project", workspace="my_workspace")
+        >>> logger.log("my_metric", 1.0, 1)
+        >>> logger.log_dict({"my_metric": 1.0}, 1)
+        >>> logger.close()
+
+    Raises:
+        ImportError: If `comet_ml` package is not installed.
+
+    Note:
+        This logger requires the comet_ml package to be installed.
+        You can install it with `pip install comet_ml`.
+        You need to set up your Comet.ml API key before using this logger.
+        You can do this by setting the COMET_API_KEY environment variable
+        or by calling `comet_ml.login()` with your API key.
+    """
+
+    def __init__(
+        self,
+        project_name: Optional[str] = None,
+        workspace: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+        log_code: bool = True,
+        **kwargs,
+    ):
+        try:
+            from comet_ml import Experiment
+        except ImportError as e:
+            raise ImportError(
+                "``comet_ml`` package not found. Please install comet_ml using `pip install comet_ml` to use CometLogger."
+                "Alternatively, use the ``StdoutLogger``, which can be specified by setting metric_logger_type='stdout'."
+            ) from e
+
+        _, self.rank = get_world_size_and_rank()
+
+        if self.rank == 0:
+            self.experiment = Experiment(
+                project_name=project_name,
+                workspace=workspace,
+                log_code=log_code,
+                **kwargs,
+            )
```

Comment:
We are moving towards a new way of initializing Comet Experiment with a new top-level function: `comet_ml.start`. This has several benefits:
* It support online and offline logging
* It support logging a new experiment or appending to an existing Experiment
* You can set name and tags directly at initialization time

I'm working right now on finalizing the documentation but I can share the reference doc for that new function already: https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment-Creation/

If there is a chance that users might want to append data to an existing experiment or create log data to an offline Experiment (to upload later), I would recommend using that instead. Please let me know, happy to provide the code corresponding code change.

Reply:
Thanks @Lothiraldan, I swapped it over to the new comet_ml.start() and verified the latest code works for me. Could you use it to make an example project which I can then link in the deep dive tutorial?

Reply:
@Lothiraldan I also added you as a collaborator to my fork, feel free to directly push changes if you would like